contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,445 @@
+        public
+        view
+        returns (bytes32)
+    {
+        bytes32 txHash = keccak256(
+            abi.encode(TX_TYPEHASH, token, to, amount, refund, gasToken, refundReceiver, _nonce)
+        );
+        return keccak256(
+            abi.encodePacked(byte(0x19), byte(0x01), domainSeparator, txHash)
+        );
+    }
+
+    /// @dev Returns start of the current time period.
+    /// @return Unix timestamp.
+    function currentStartTime()
+        public
+        view
+        returns (uint)
+    {
+        return now - (now % timePeriod);
+    }
+
+    function checkTransferLimits(address token, uint256 amount)
+        internal
+        view
+        returns (bool, uint256, uint256)
+    {
+        TransferLimit storage transferLimit = transferLimits[token];
+        // Transfer + previous expenditure shouldn't exceed limit specified for token.
+        if (transferLimit.spent.add(amount) > transferLimit.limit) {
+            return (false, 0, 0);
+        }
+
+        // If a global cap is set, transfer amount + value of all
+        // previous expenditures (for all tokens) shouldn't exceed global limit.
+        return isUnderGlobalCap(token, amount);
+    }
+
+    function isUnderGlobalCap(address token, uint256 amount)
+        internal
+        view
+        returns (bool, uint256, uint256)
+    {
+        if (globalWeiCap == 0 && globalDaiCap == 0) {
+            return (true, 0, 0);
+        }
+
+        // Calculate value in ether.
+        uint256 ethNum;
+        uint256 ethDen;
+        (ethNum, ethDen) = getEthAmount(token, amount);
+
+        // Convert ether to wei
+        uint256 weiAmount = ethNum.mul(10**18).div(ethDen);
+        uint256 weiSpent = totalWeiSpent.add(weiAmount);
+        if (globalWeiCap > 0 && weiSpent > globalWeiCap) {
+            return (false, 0, 0);
+        }
+
+        uint256 daiSpent;
+        if (globalDaiCap != 0) {
+            // Calculate value in dai.
+            uint256 daiNum;
+            uint256 daiDen;
+            (daiNum, daiDen) = getDaiAmount(ethNum, ethDen);
+
+            uint256 daiAmount = daiNum.div(daiDen);
+            daiSpent = totalDaiSpent.add(daiAmount);
+            if (daiSpent > globalDaiCap) {
+                return (false, 0, 0);
+            }
+        }
+
+        return (true, weiSpent, daiSpent);
+    }
+
+    function isPeriodOver()
+        internal
+        returns (bool)
+    {
+        if (rolling && now > lastStartTime + timePeriod) {
+            lastStartTime = now;
+            return true;
+        } else if (!rolling && currentStartTime() > lastStartTime) {
+            lastStartTime = currentStartTime();
+            return true;
+        }
+
+        return false;
+    }
+
+    function updateTransferLimits(address token, uint256 amount, uint256 weiSpent, uint256 daiSpent)
+        internal
+    {
+        transferLimits[token].spent = transferLimits[token].spent.add(amount);
+        if (weiSpent > 0 && weiSpent > totalWeiSpent) {
```

Comment:
Can you comment this code pls. This might be potentially confusing. (I assume this is to check if the totalWeiSpent has increased we should update it, since 0 indicates that we should not update it)